src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -370,13 +370,48 @@ public void It_includes_internals_visible_to()
             AssemblyInfo.Get(assemblyPath)["InternalsVisibleToAttribute"].Should().Be("Tests");
         }
 
+        private static string _cachedCurrentTFM = null;
+
+        private static string CurrentTFM
+        {
+            get
+            {
+                if (_cachedCurrentTFM == null)
+                {
+                    var logger = new StringTestLogger();
+                    var testAssetsManager = new TestAssetsManager(logger);
+                    TestDirectory testDirectory = testAssetsManager.CreateTestDirectory();
+                    string path = testDirectory.Path;
+                    var cmd = new DotnetCommand(logger)
+                        .WithWorkingDirectory(path)
+                        .Execute("new", "console");
+
+                    string lines = File.ReadAllText(Path.Combine(path, "CurrentTFM.csproj"));
+                    string startTargetFrameworkText = @"<TargetFramework>";
+                    string endTargetFrameworkText = @"</TargetFramework>";
```

Comment:
It would be better to use XML APIs to parse the project XML.  Here's an example:

https://github.com/dotnet/installer/blob/44a674908e37892dc1833f8cc7f8f0d9ec48eeb3/test/EndToEnd/ProjectBuildTests.cs#L407-L410